std/variant.d

```diff
@@ -3097,3 +3101,26 @@ if (isAlgebraic!VariantType && Handler.length > 0)
 
     static assert(is(Outer.AllowedTypes == AliasSeq!(Inner, Outer*)));
 }
+
+// https://issues.dlang.org/show_bug.cgi?id=15615
+@system unittest
+{
+    alias Value = Algebraic!(long, double);
+
+    const long foo = 123L;
+    long bar = foo;
+    Value baz = Value(foo);
+    assert(baz == 123L);
+
+    // Should not allow for const transitive violation
+    static struct Wrapper
+    {
+        long* ptr;
+    }
+
+    alias VariantWrapper = Algebraic!(Wrapper, double);
+
+    long l = 2;
+    const cw = Wrapper(&l);
+    static assert(!__traits(compiles, VariantWrapper(cw)), "Stripping `const` from type with indirection!");
```

Comment:
If this triggered, I'm worried the error message would mislead. If I were to get a message like this, I'd think that the unit test is accidently trying to strip `const` and `Algebraic` is trying to prevent it. Would "std.variant.Variant is wrongly allowing stripping `const` from type with indirection!" be better?